pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -382,7 +384,7 @@ def _check_device_config_and_set_final_flags(
             devices, num_processes, gpus, ipus, tpu_cores
         )
 
-        if self._devices_flag in ([], 0, "0", "0,"):
```

Comment:
Is this a breaking change?

Reply:
Nope, it is actually fixing an issue where `Trainer(accelerator="gpu", devices="0,")` was not selecting the first gpu.

Reply:
We should add a test then to avoid regressions

Reply:
confirmed, this is only a problem on master. 
Do you think we could pull the fix out of this PR?

Reply:
> or a parametrization to an existing test


Yup, https://github.com/PyTorchLightning/pytorch-lightning/blob/5f1a1e40487751c5da6a501743e42aa1d1e86a3d/tests/trainer/test_trainer_cli.py#L166